### PR TITLE
Pom madness

### PR DESCRIPTION
**Pull request by norwood**
dependent on: https://github.com/confluentinc/common/pull/43

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -43,13 +47,11 @@
 
     <properties>
         <confluent.maven.repo>http://packages.confluent.io/maven/</confluent.maven.repo>
-        <confluent.common.version>3.2.1-SNAPSHOT</confluent.common.version>
-        <slf4j.version>1.7.21</slf4j.version>
+        <confluent.common.version>${project.version}</confluent.common.version>
```
Comment (ewencp) [inline pom.xml]: Should we just get rid of this variable entirely since it's the same as the project version anyway, or do you still see value in having the level of indirection? Perhaps to make any subsequent transition to using separate version numbers for different components in the future?

I'm mildly embarrassed that just reusing the project version didn't occur to me. We'd probably significantly simplify the version bumping tooling by doing this across the board, at least as long as we stick with fully coordinated version numbers.

File: pom.xml [hunk 2]
```diff
@@ -43,13 +47,11 @@
 
     <properties>
         <confluent.maven.repo>http://packages.confluent.io/maven/</confluent.maven.repo>
-        <confluent.common.version>3.2.1-SNAPSHOT</confluent.common.version>
-        <slf4j.version>1.7.21</slf4j.version>
```
Comment (ewencp) [inline pom.xml]: Is this just the starting point? Or will other repos use a different parent pom (e.g. this one)? Things like jackson and the `kafka.scala.version` seem like they could be shared to avoid potential conflicts (e.g. differing jackson versions across connectors can be an issue).

File: pom.xml [hunk 3]
```diff
@@ -64,28 +66,69 @@
...
+                <artifactId>jackson-annotations</artifactId>
+                <version>${jackson.version}</version>
+            </dependency>
+        </dependencies>
+    </dependencyManagement>
 
     <build>
         <plugins>
             <plugin>
-                <groupId>org.apache.maven.plugins</groupId>
```
Comment (ewencp) [inline pom.xml]: Are build plugins inherited from the parent pom such that this is no longer required? This is a really important requirement for us so I don't want to mess it up (though I think our system tests would catch any issues since we're careful about only installing the earliest jdk we support).

File: test/pom.xml [hunk 4]
```diff
@@ -24,7 +24,7 @@
         <dependency>
             <groupId>junit</groupId>
             <artifactId>junit</artifactId>
-            <version>${junit.version}</version>
+            <scope>compile</scope>
```
Comment (ewencp) [inline test/pom.xml]: I don't understand why this change is needed? Isn't `compile` the default scope?

Comment (norwood) [inline test/pom.xml]: it is not the default since i moved the dep up to dependency manager in a parent pom. this needs to be reset to compile since test package is making a deployable for others to load

File: pom.xml (hunk 2 above)
Comment (norwood) [inline pom.xml]: yeah, starting point. 

we could definitely have things like schema-registry and rest-proxy extend this pom so they can get jetty version (since i doubt anyone else needs that). and move jackson or something else shared up... though we can/will run in to conflicts there.

File: pom.xml (hunk 1 above)
Comment (norwood) [inline pom.xml]: i think its fine to leave like this now. say at some point we move off of snapshot to independently versioned stuff

File: pom.xml (hunk 3 above)
Comment (norwood) [inline pom.xml]: yes, here is the effective pom https://gist.github.com/norwood/16971fb2709d62997b75614cb1254bb4

**Comment by norwood** [comment]
retest this please

**Comment by norwood** [comment]
retest this please
